File: rb/features.rb
# Parkleitsystem SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module ParkleitsystemFeatures
  def self.make_feature(name)
    case name
    when "base"
      ParkleitsystemBaseFeature.new
    when "ratelimit"
      ParkleitsystemRatelimitFeature.new
    when "retry"
      ParkleitsystemRetryFeature.new
    when "test"
      ParkleitsystemTestFeature.new
    when "timeout"
      ParkleitsystemTimeoutFeature.new
    else
      ParkleitsystemBaseFeature.new
    end
  end
end

File: py/features.py
# Parkleitsystem SDK feature factory

from feature.base_feature import ParkleitsystemBaseFeature
from feature.test_feature import ParkleitsystemTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ParkleitsystemBaseFeature(),
        "test": lambda: ParkleitsystemTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
